File: StokTakipFE/src/app/services/serviceClasses/ApiStockProductItem.ts
import { ApiProductDefinitionResponse } from './ApiProductDefinitionResponse';

export interface ApiStockProductItem {
    serialNo: string;
    productName: ApiProductDefinitionResponse;
    stockProductName: string;
    quantity: string;
    department: string;
}

export class ApiStockProductItem implements ApiStockProductItem {
    serialNo: string;
    productName: ApiProductDefinitionResponse;
    stockProductName: string;
    quantity: string;
    department: string;

    constructor(o: ApiStockProductItem) {
        if (!o || !o.serialNo) {
            return;
        }
        this.serialNo = o.serialNo;
        this.productName = o.productName;
        this.stockProductName = o.stockProductName;
        this.quantity = o.quantity;
        this.department = o.department;
    }
}

File: StokTakip/StokTakip/ApiClasses/ApiStockProductItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StokTakip.ApiClasses
{
    public class ApiStockProductItem
    {
        public string serialNo { get; set; }
        public ApiProductDefinitionResponse productName { get; set; }
        public int quantity { get; set; }
        public string department { get; set; }
    }
}
